### PR TITLE
New Feature: #12686 SoftmaxCrossEntropyWithLogits gradient function

### PR DESCRIPTION
**Pull request by pbanavara**
Committing in a new PR as the old PR has too many commit files

### CODE REVIEW

**Comment by pbanavara** [comment]
@suharshs This is the new PR taking over from the old one. Please let me know if there are any issues merging this.

**Review by suharshs — APPROVED** [review]

**Comment by pbanavara** [comment]
Awesome ! Hey thank you everyone for helping me in pulling this together. I took an awfully long time to get this done. This is my first ever PR to Tensorflow so it means a lot ! Thanks again.
